src/components/form-elements/draft-js-mention-selector/DraftMentionItem.js

```diff
@@ -2,16 +2,34 @@
 import * as React from 'react';
 import { ContentState } from 'draft-js';
 
+const MENTION_REGEX = /([@＠﹫])\[([0-9]+):([^\]]+)]/i;
+
 type Props = {
     children: React.Node, // React.ChildrenArray<React.Element<any>> unsupported by babel-plugin-flow-react-proptypes
     contentState: ContentState,
+    decoratedText: string,
     entityKey: string,
 };
 
-const DraftMentionItem = ({ contentState, entityKey, children }: Props) => {
-    const { id } = contentState.getEntity(entityKey).getData();
+const DraftMentionItem = ({ contentState, entityKey, children, decoratedText }: Props) => {
+    let id;
+    let content;
+
+    // if the block was created in the editing session it will have an entityKey
+    if (entityKey) {
+        id = contentState.getEntity(entityKey).getData().id;
+        content = children;
+    } // if the block is parsed from plain text it will look like @[id:name]
+    else {
+        const mentionMatch = decoratedText.match(MENTION_REGEX);
+        if (mentionMatch) {
+            const [, triggerSymbol, userId, name] = mentionMatch;
+            id = userId;
+            content = `${triggerSymbol}${name}`;
```

Comment:
Decorators shouldn't do this - the content is immutable; the decorator should only be wrapping `props.children`